src/main/scala/za/co/absa/commons/lang/extensions/TraversableExtension.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.commons.lang.extensions
+
+object TraversableExtension {
+
+  implicit class TraversableOps[A <: Traversable[_]](val xs: A) extends AnyVal {
+
+    def asOption: Option[A] = if (xs.isEmpty) None else Some(xs)
```

Comment:
Isn't this covered in [`NonOptionExtension`]([NonOptionExtension.scala](https://github.com/AbsaOSkjk,.S/commons/pull/108/files#diff-154e0e72b8f99fdf0051419d6dec1408e403c782bf6b5fbff3943ef3565cab7b))?

Reply:
No this is different. The one in `NonOptionExtension` would return `Some(List())` and the one here `None` if called on empty list. I committed an update to above method that allows being called on nulls + added a description of the method.

Reply:
it would also be good to cover added boundary conditions with tests

Reply:
Added tests for TraversableExtension

Reply:
Thanks for the explanation, of course it's different. Cool with the added comment.

Reply:
An good idea with the null check 👍